spec/double_entry/reporting/aggregate_spec.rb

```diff
@@ -42,6 +44,23 @@ module Reporting
         Aggregate.new(:sum, :savings, :bonus, TimeRange.make(:year => 2009, :month => 9)).amount
         Aggregate.new(:sum, :savings, :bonus, TimeRange.make(:year => 2009, :month => 10)).amount
         expect(LineAggregate.count).to eq 2
+        expect(LineAggregate).to have_received(:aggregate).twice
+      end
+
+      it 'calculates a new aggregate when partner_account is specified' do
+        Aggregate.new(:sum, :savings, :bonus, TimeRange.make(:year => 2009, :month => 9)).amount
+        Aggregate.new(:sum, :savings, :bonus, TimeRange.make(:year => 2009, :month => 9), partner_account: :test).amount
+        Aggregate.new(:sum, :savings, :bonus, TimeRange.make(:year => 2009, :month => 10)).amount
+        expect(LineAggregate.count).to eq 3
+        expect(LineAggregate).to have_received(:aggregate).exactly(3).times
+      end
+
+      it "only stores an aggregate including partner_account once if it's requested more than once" do
+        Aggregate.new(:sum, :savings, :bonus, TimeRange.make(:year => 2009, :month => 9), partner_account: :test).amount
+        Aggregate.new(:sum, :savings, :bonus, TimeRange.make(:year => 2009, :month => 9), partner_account: :test).amount
+        Aggregate.new(:sum, :savings, :bonus, TimeRange.make(:year => 2009, :month => 10), partner_account: :test).amount
+        expect(LineAggregate.count).to eq 2
+        expect(LineAggregate).to have_received(:aggregate).twice
       end
```

Comment:
I'd like to see some concrete tests on the figures for partner_account specified and partner_account=nil.

And can we correct the spelling of 'separate' below please :-P